test/csv/interface/test_read_write.rb

```diff
@@ -24,6 +24,20 @@ def test_filter
     CSV
   end
 
+  def test_filter_with_headers
+    input = <<-CSV
+Name,Value
+foo,0
+bar,1
+baz,2
+    CSV
+    output = ""
+    CSV.filter(input, output, headers: true, write_headers: true) do |row|
```

Comment:
We should use `output_write_headers:` here. If we use `write_headers:`, `input` is also rewritten.